src/commands/lambda/flare.ts

```diff
@@ -771,6 +784,75 @@ export const sleep = async (ms: number) => {
   await new Promise((resolve) => setTimeout(resolve, ms))
 }
 
+/**
+ * Get the framework used based on the files in the directory
+ * @returns the framework used or undefined if no framework is found
+ */
+export const getFramework = () => {
+  const files = new Set(fs.readdirSync(process.cwd()))
+
+  // Check for serverless framework
+  const serverlessFiles = ['serverless.yml', 'serverless.yaml']
+  for (const file of serverlessFiles) {
+    if (files.has(file)) {
+      return 'Serverless Framework'
+    }
+  }
+
+  // Check for AWS CDK
+  if (files.has('cdk.json')) {
+    return 'AWS CDK'
+  }
+
+  // Check for AWS CloudFormation
+  const cloudFormationFiles = ['template.yml', 'template.yaml']
+  for (const file of cloudFormationFiles) {
+    if (files.has(file)) {
+      return 'AWS CloudFormation'
+    }
+  }
+
+  return 'Unknown'
+}
+
+/**
+ * Generate the insights file
+ * @param insightsFilePath path to the insights file
+ * @param isDryRun whether or not this is a dry run
+ * @param config Lambda function configuration
+ */
+export const generateInsightsFile = (insightsFilePath: string, isDryRun: boolean, config: FunctionConfiguration) => {
```

Comment:
Missing:
1. FunctionName,
2. ARN,
3. Handler – or equivalent if Image
4. Environment Variables,
5. Timeout,
6. Memory Size,
7. Architecture.

This file does not contain any separation between what's CLI run data and what's customer's AWS Lambda Data.

I'd suggest creating a markdown template, the current approach is just a list in a markdown file.

Reply:
Thanks, I added the missing fields and separated the CLI data vs Lambda config data